randovania/inventory_server/inventory_server.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+import threading
+import websockets
+import logging
+import json
+import enum
+from PySide6 import QtWidgets
+from randovania.game_connection.game_connection import GameConnection
+from randovania.game_connection.connection_base import Inventory, GameConnectionStatus
+from randovania.interface_common.options import Options
+
+class InventoryServerStatus(enum.Enum):
+    STOPPED = "Stopped"
+    STARTING = "Starting..."
+    RUNNING = "Running"
+    STOPPING = "Stopping..."
+    ERROR = "ERROR: "
+
+class InventoryServer:
+
+    _logger: logging.Logger
+    _thread: threading.Thread
+    _game_connection: GameConnection
+    _options: Options
+    _status: InventoryServerStatus
+    _status_change_callbacks: list = []
+    _error: Exception | None = None
+
+    @property
+    def is_running(self) -> bool:
+        return self._is_running
+
+    def __init__(self, game_connection: GameConnection, options: Options):
+        self._game_connection = game_connection
+        self._options = options
+        self._set_status(InventoryServerStatus.STOPPED)
+        self._setup_logging()
+        self._autostart_server()
+    
+    def _set_status(self, status: InventoryServerStatus, error: Exception | None = None):
+        self._error = error
+        self._status = status
+        self._call_status_change_callbacks()
+    
+    def _call_status_change_callbacks(self):
+        for callback in self._status_change_callbacks:
+            callback(self._status)
+    
+    def _setup_logging(self):
+        self._logger = logging.getLogger(type(self).__name__)
+
+    def _autostart_server(self):
+        if self._options.inventory_server_autostart:
+            self.start()
+            self._logger.info("Inventory server autostarted, running on port " + str(self._options.inventory_server_port))
+
+    def start(self):
+        self._set_status(InventoryServerStatus.STARTING)
+        self._thread = threading.Thread(target=asyncio.run, args=(self._run_websocket_server(),), daemon=True)
+        self._thread.start()
+    
+    async def _run_websocket_server(self):
+        try:
+            async with websockets.serve(self._respond_to_ws_message, "localhost", int(self._options.inventory_server_port)) as ws:
+                self._ws = ws
+                self._set_status(InventoryServerStatus.RUNNING)
+                await ws.wait_closed()
+        except Exception as e:
```

Comment:
What kind of exceptions can happen during server start? Should we restart the server in case of errors?
There likely should be some logging calls to record the errors properly.